### PR TITLE
Fixed filter search fields and query sorting conditions. Further improve overall redis client features

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR addresses the following:
- search SORTBY clause is asking for the sort wrong order
- filtering conditions for numeric/tags are now properly addressed
- adds the ability to run in cluster mode 
- uses latest dialect features on redisearch

### CODE REVIEW

**Comment by generall** [comment]
hey @filipecosta90 thanks for the pointers!

Is Redis Cluster search open-sourse now? I might have missed that

**Comment by filipecosta90** [comment]
> hey @filipecosta90 thanks for the pointers!
> 
> Is Redis Cluster search open-sourse now? I might have missed that

it is :) I'll describe on this issue how to use it until EOD

**Comment by filipecosta90** [comment]
@generall as I was saying the redis search capabilities where open sourced in 2020 as described in https://redis.com/blog/introducing-redisearch-2-0/ . 
From that point forward, it is all present in https://github.com/RediSearch/RediSearch and for you to build the OSS cluster artifact you just `make build COORD=1` as visible on the makefile https://github.com/RediSearch/RediSearch/blob/master/Makefile#L141

**Comment by generall** [comment]
> @generall as I was saying the redis search capabilities where open sourced in 2020 as described in https://redis.com/blog/introducing-redisearch-2-0/ . 


I mean the distributed version, multiple nodes

**Comment by filipecosta90** [comment]
> > @generall as I was saying the redis search capabilities where open sourced in 2020 as described in https://redis.com/blog/introducing-redisearch-2-0/ .
> 
> I mean the distributed version, multiple nodes

Yes. 

In the blog I sent above its clearly stated in:

> OSS cluster support for open source Redis

> We didn’t want to limit increasing the scalability capabilities to only Redis Enterprise users, so we added support for scaling a single index over multiple shards with the open source Redis cluster API. Previously, a single RediSearch index, and its documents, had to reside on a single shard. This meant that dataset size and throughput for OSS Redis was bound to what a single Redis process could handle. Redis Enterprise offered the ability to distribute documents in a clustered database and aggregate the results at query time. This fan-out and aggregation is handled by a component called the “coordinator” that is now also [publicly available ](https://github.com/RediSearch/RediSearch/coord)under the [Redis Source Available License](https://redis.com/legal/licenses/) so it will work with open source Redis clusters as well as Redis Enterprise. The result is the most scalable version of RediSearch yet.

The explanation on how to run on multi-node is in https://github.com/RediSearch/RediSearch/tree/master/coord 
TLDR, yes, multi-node OSS as described and documented above.
